### DISCUSSION TITLE
Can't commit changes

### DISCUSSION THREAD

**Discussion by smita-jh** [community/community#120420] [Codespaces]
### Select Topic Area

Bug

### Body

PS D:\M.Tech\SK\Javascript\GitDemo\Project2> git commit -m"First Files"
Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'DELL@DESKTOP-0P26MKM.(none)')

**Comment by eddysantiagoo** [comment]
Hi @smita-jh The error message you're seeing is because Git needs to know your username and email address before it can commit your changes. This is because every Git commit uses this information to associate each commit with a specific author.

You can fix this by running the following commands in your terminal, replacing `"yourgithubemail@example.com"` with your email address and "Your Name" with your name:

```bash
git config --global user.email "yourgithubemail@example.com"
git config --global user.name "Your Name"
```

These commands set your email and name globally for all repositories on your system, after running these commands, you should be able to commit your changes without any issues :)


If this answer helps you, please mark it as the correct answer
